kodilla-testing2/src/test/java/testing2/crudapp/CrudAppTestSuite.java

```diff
@@ -0,0 +1,132 @@
+package testing2.crudapp;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+import testing2.config.WebDriverConfig;
+
+import java.util.Random;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+public class CrudAppTestSuite {
+    public static final String BASE_URL = "https://domibroz.github.io";
+    private WebDriver driver;
+    private Random generator;
+
+    @Before
+    public void initTests() {
+        driver = WebDriverConfig.getDriver(WebDriverConfig.CHROME);
+        driver.get(BASE_URL);
+        generator = new Random();
+    }
+
+    @After
+    public void cleanUpAfterTest() {
+        driver.close();
+    }
+
+    private String createCrudAppTestTask() throws InterruptedException {
+        final String XPATH_TASK_NAME = "/html/body/main/section[1]/form/fieldset[1]/input";
+        final String XPATH_TASK_CONTENT = "/html/body/main/section[1]/form/fieldset[2]/textarea";
+        final String XPATH_ADD_BUTTON = "/html/body/main/section[1]/form/fieldset[3]/button";
+        String taskName = "Task number " + generator.nextInt(100000);
+        String taskContent = taskName + " content";
+
+        while (!driver.findElement(By.xpath(XPATH_TASK_NAME)).isDisplayed()) ;
+
+        WebElement name = driver.findElement(By.xpath(XPATH_TASK_NAME));
+        name.sendKeys(taskName);
+
+        WebElement content = driver.findElement(By.xpath(XPATH_TASK_CONTENT));
+        content.sendKeys(taskContent);
+
+        WebElement addButton = driver.findElement(By.xpath(XPATH_ADD_BUTTON));
+        addButton.click();
+        Thread.sleep(2000);
+
+        return taskName;
+    }
+
+    private void sendTestTaskToTrello(String taskName) throws InterruptedException {
+        driver.navigate().refresh();
+
+        while (!driver.findElement(By.xpath("//select[1]")).isDisplayed()) ;
```

Comment:
you should create helper method, waitFor which performs given check, in this example if element is displayed, but between checks it goes sleep (for example 100 ms) - thanks to that CPU consumption will be lower. Additionally, method should have reasonable timeout - how long we want to perform checks.